### PR TITLE
Event handlers

### PR DESCRIPTION
**Pull request by jeton-th**

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,15 +1,31 @@
 import React from 'react';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
-import calculate from '../logic/calculate';
 import './style/App.css';
+import calculate from '../logic/calculate';
 
 class App extends React.Component {
+  constructor(props) {
```
Comment (rvvergara) [inline src/components/App.js]: There's a better way to write our classes, this is without using a constructor and also with the use of arrow functions:

```javascript
clas App extends React.Component {
  state = {
      total: null,
      next: null,
      operation: null,
    };
  // the ff arrow function format saves us from the ugly "bind" of 'this':
  handleClick = (buttonName) => {
    const newState = calculate(this.state, buttonName.target.innerHTML);
    this.setState(() => ({ ...newState }));
  }
}
```

Further, the `handleClick` can use of the [Functional format](https://www.freecodecamp.org/news/functional-setstate-is-the-future-of-react-374f30401b6b/) of setState:

```javascript
handleClick = buttonName => this.setState(prevState => (
    calculate(prevState, buttonName)
  ))
```

File: src/components/App.js [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(buttonName) {
```
Comment (rvvergara) [inline src/components/App.js]: See my comment above on how to use arrow functions and functional setState format for `handleClick`.

File: src/components/App.js [hunk 3]
```diff
@@ -1,15 +1,31 @@
...
+    const newState = calculate(this.state, buttonName.target.innerHTML);
```
Comment (rvvergara) [inline src/components/App.js]: We can simply use `buttonName` here and not have to do `buttonName.target.innerHTML`.

The problem happens with what we assigned to the `onClick` prop in `Button.js`. I won't give you the full solution here and I want you to figure out the problem in `Button.js`. But the point is we can simply use `buttonName` without having to get to the `target.innerHTML`

File: src/components/Button.js [hunk 4]
```diff
@@ -1,24 +1,26 @@
...
+  const buttonsStyle = {
+    border: '1px solid lightgray',
+    fontSize: 30,
     backgroundColor: color,
     width: (wide) ? '50%' : '25%',
   };
 
   return (
-    <button type="button" style={buttonsStyle}>{name}</button>
+    <button type="button" style={buttonsStyle} onClick={clickHandler}>{name}</button>
```
Comment (rvvergara) [inline src/components/Button.js]: There's a serious issue in the `onClick` prop here. Let's figure out what's the problem in relation to my comment in `App.js` about not having to do `buttonName.target.innerHTML`.

**Comment by dipto0321** [comment]
HI @jeton-th ,

Good job. According to last feedback you solved most of the refactoring and issues except # 7. Here is look at the picture below.

 After fixing this issue you can merge it. This is from me 👍 . You can also share this project live review link to get some real feedback from other people in full-time-students slack channel. You can think it like a product survey 😄 

Thanks 🙏

**Pull request merged**